public/app/core/components/PageHeader/PageHeader.tsx

```diff
@@ -71,86 +71,92 @@ const Navigation = ({ children }: { children: NavModelItem[] }) => {
-    if (!title && (!breadcrumbs || breadcrumbs.length === 0)) {
-      return null;
-    }
+  const main = model.main;
+  const children = main.children;
 
-    if (!breadcrumbs || breadcrumbs.length === 0) {
-      return <h1 className="page-header__title">{title}</h1>;
-    }
+  return (
+    <div className={styles.headerCanvas}>
+      <div className="page-container">
+        <div className="page-header">
+          {renderHeaderTitle(main)}
+          {children && children.length && <Navigation>{children}</Navigation>}
+        </div>
+      </div>
+    </div>
+  );
+};
 
-    const breadcrumbsResult = [];
-    for (const bc of breadcrumbs) {
-      if (bc.url) {
-        breadcrumbsResult.push(
-          <a className="text-link" key={breadcrumbsResult.length} href={bc.url}>
-            {bc.title}
-          </a>
-        );
-      } else {
-        breadcrumbsResult.push(<span key={breadcrumbsResult.length}> / {bc.title}</span>);
-      }
-    }
-    breadcrumbsResult.push(<span key={breadcrumbs.length + 1}> / {title}</span>);
+function renderHeaderTitle(main: NavModelItem) {
+  const iconClassName =
+    main.icon === 'grafana'
+      ? css`
+          margin-top: 12px;
+        `
+      : css`
+          margin-top: 14px;
+        `;
 
-    return <h1 className="page-header__title">{breadcrumbsResult}</h1>;
-  }
+  return (
+    <div className="page-header__inner">
+      <span className="page-header__logo">
+        {main.icon && <Icon name={main.icon as IconName} size="xxxl" className={iconClassName} />}
+        {main.img && <img className="page-header__img" src={main.img} alt={`logo of ${main.text}`} />}
+      </span>
+
+      <div className="page-header__info-block">
+        {renderTitle(main.text, main.breadcrumbs ?? [])}
+        {main.subTitle && <div className="page-header__sub-title">{main.subTitle}</div>}
+      </div>
+    </div>
+  );
+}
 
-  renderHeaderTitle(main: NavModelItem) {
-    const iconClassName =
-      main.icon === 'grafana'
-        ? css`
-            margin-top: 12px;
-          `
-        : css`
-            margin-top: 14px;
-          `;
+function renderTitle(title: string, breadcrumbs: NavModelBreadcrumb[]) {
+  if (!title && (!breadcrumbs || breadcrumbs.length === 0)) {
+    return null;
+  }
 
+  if (!breadcrumbs || breadcrumbs.length === 0) {
     return (
-      <div className="page-header__inner">
-        <span className="page-header__logo">
-          {main.icon && <Icon name={main.icon as IconName} size="xxxl" className={iconClassName} />}
-          {main.img && <img className="page-header__img" src={main.img} alt={`logo of ${main.text}`} />}
-        </span>
-
-        <div className="page-header__info-block">
-          {this.renderTitle(main.text, main.breadcrumbs ?? [])}
-          {main.subTitle && <div className="page-header__sub-title">{main.subTitle}</div>}
-        </div>
-      </div>
+      <h1 className="page-header__title" data-testid="page-title">
```

Comment:
`data-testid` shouldn't be necessary here. You'd be able to query this element via `screen.getByRole('heading', {name: [the string that title evaluates to]})`.